dlls/ff/ff_player.cpp

```diff
@@ -5405,14 +5496,11 @@ int CFFPlayer::OnTakeDamage(const CTakeDamageInfo &inputInfo)
 	//		pAttacker->AddFortPoints(10,true);
 	//}
 
-	// if it's a pyro, they take half damage
-	// AfterShock: pyros now take full damage. They can't be lit on fire tho, and thats a big enough differnce!
-	/*
-	if ( GetClassSlot() == CLASS_PYRO && info.GetDamageType()&DMG_BURN )
+	// if it's a pyro shooting themself (i.e. the IC) they take less damage
+	if ( GetClassSlot() == CLASS_PYRO && info.GetDamageType()&DMG_BURN && (info.GetInflictor() == this || info.GetAttacker() == this))
 	{
-		info.SetDamage(info.GetDamage()/1.3); //Instead of taking 50% damage, pyros now take 75% - AfterShock
+		info.SetDamage(info.GetDamage() * FFDEV_PYRO_IC_MULTIPLIER);

```

Comment:
This is duplicating functionality that also exists in ff_gamerules.cpp. See: https://github.com/fortressforever/fortressforever/blob/beta/game_shared/ff/ff_gamerules.cpp#L1584-L1587